macros/edr/dbt_artifacts/upload_artifacts_to_table.sql

```diff
@@ -9,7 +15,8 @@
     {%- set flatten_artifacts_len = flatten_artifact_dicts | length %}
     {% if flatten_artifacts_len > 0 %}
         {{ elementary.debug_log('Inserting ' ~ flatten_artifacts_len ~ ' rows to table ' ~ table_relation) }}
-        {% do elementary.insert_dicts(table_relation, flatten_artifact_dicts, elementary.get_config_var('dbt_artifacts_chunk_size'), should_commit) %}
+        {% do dbt.truncate_relation(table_relation) %}
```

Comment:
This is not atomic, but I think it's inevitable given that we're using batched inserts.